### PR TITLE
More hidden friend operators

### PR DESCRIPTION
**Pull request by jfalcou**
The == != > < >= operators are now friends.

### CODE REVIEW

File: include/eve/detail/function/simd/arm/neon/friends.hpp [hunk 1]
```diff
@@ -79,6 +79,121 @@ namespace eve::detail
...
+  EVE_FORCEINLINE as_logical_t<wide<T, N, ABI>> self_neq(wide<T,N,ABI> v, wide<T,N,ABI> w) noexcept
+  {
+    return !(v == w);
+  }
+
+  //================================================================================================
+  // operator!= implementation
+  //================================================================================================
+  template<typename T, typename N, arm_abi ABI>
+  EVE_FORCEINLINE as_logical_t<wide<T, N>> self_less( wide<T, N, ABI> v
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/arm/neon/friends.hpp:98]: `al_logical<wide<T, N, ABI>>`?

File: include/eve/detail/function/simd/arm/neon/friends.hpp [hunk 2]
```diff
@@ -79,6 +79,121 @@ namespace eve::detail
...
+    else  if constexpr( cat == category::uint8x16 ) return vcltq_u8(v, w);
+    else  if constexpr( cat == category::float32x4) return vcltq_f32(v, w);
+    else  if constexpr( cat == category::int32x2  ) return vclt_s32(v, w);
+    else  if constexpr( cat == category::int16x4  ) return vclt_s16(v, w);
+    else  if constexpr( cat == category::int8x8   ) return vclt_s8(v, w);
+    else  if constexpr( cat == category::uint32x2 ) return vclt_u32(v, w);
+    else  if constexpr( cat == category::uint16x4 ) return vclt_u16(v, w);
+    else  if constexpr( cat == category::uint8x8  ) return vclt_u8(v, w);
+    else  if constexpr( cat == category::float32x2) return vclt_f32(v, w);
+#if defined(__aarch64__)
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/arm/neon/friends.hpp:122]: You have the proper condition for it, why defines?

File: include/eve/detail/function/simd/arm/neon/friends.hpp [hunk 3]
```diff
@@ -79,6 +79,121 @@ namespace eve::detail
...
+    else  if constexpr( cat == category::float64x1) return vcgt_f64(v, w);
+    else  if constexpr( cat == category::int64x1)   return vcgt_s64(v, w);
+    else  if constexpr( cat == category::uint64x1)  return vcgt_u64(v, w);
+    else  if constexpr( cat == category::float64x2) return vcgtq_f64(v, w);
+    else  if constexpr( cat == category::int64x2)   return vcgtq_s64(v, w);
+    else  if constexpr( cat == category::uint64x2)  return vcgtq_u64(v, w);
+#else
+    else  if constexpr( sizeof(T) == 8 )
+      return map([]<typename E>(E const& e, E const& f){ return as_logical_t<E>(e > f); }, v, w);
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/arm/neon/friends.hpp:164]: This is not the same as for `<` Why?

File: include/eve/function/is_greater_equal.hpp [hunk 4]
```diff
@@ -10,25 +10,21 @@
...
-#  include <eve/module/real/core/function/regular/simd/arm/neon/is_greater_equal.hpp>
-#endif
 
+  namespace detail
+  {
+    template<value T, value U>
+    EVE_FORCEINLINE auto is_greater_equal_(EVE_SUPPORTS(cpu_), T const &a, U const &b) noexcept
+    {
+      if constexpr( scalar_value<T> && scalar_value<U> )  return as_logical_t<T>(a >= b);
+      else                                                return a <= b;
```
Comment (DenisYaroshevskiy) [inline include/eve/function/is_greater_equal.hpp:27]: Doesn't seem right.

**Review by DenisYaroshevskiy — COMMENTED** [review]
Just passing by

File: include/eve/detail/function/simd/arm/neon/friends.hpp (hunk 3 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/neon/friends.hpp:164]: becaue I didnt updated <. This version is better as I don't reuse th elambda anywhere else

File: include/eve/function/is_greater_equal.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/function/is_greater_equal.hpp:27]: yup stupid copypasta

File: include/eve/detail/function/simd/arm/neon/friends.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/neon/friends.hpp:98]: oversight indeed

File: include/eve/detail/function/simd/arm/neon/friends.hpp (hunk 2 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/neon/friends.hpp:122]: the #if need to go when we will have neon128_v8.
But I want to do this change later in another PR.

**Pull request merged**